gRPC/Client Examples/EvokedPotentialsApp/BICManager.cs

```diff
@@ -168,6 +175,43 @@ public bool BICConnect()
             Console.WriteLine("Connecting to implantable device.");
             var connectDeviceReply = deviceClient.ConnectDevice(new ConnectDeviceRequest() { DeviceAddress = DeviceName, LogFileName = "./deviceLog.txt" });
 
+            // Print out impedances of electrodes
+            for (uint channelNum = 0; channelNum < numSensingChannelsDef; channelNum++)
+            {
+                bicGetImpedanceReply chanImpedValue = deviceClient.bicGetImpedance(new bicGetImpedanceRequest() { DeviceAddress = DeviceName, Channel = channelNum });
+                Console.WriteLine("CH " + (channelNum + 1).ToString() + ": " + chanImpedValue.Success);
+                if (chanImpedValue.Success == "success")
+                {
+                    // add impedance value and units to buffer
+                    impedBuffer.Add(chanImpedValue.ChannelImpedance.ToString() + " " + chanImpedValue.Units);
+                }
+                else
+                {
+                    // note error in retrieiving impedance 
+                    impedBuffer.Add(chanImpedValue.Success);
+                }
+            }
+
+            // Check that impedances can be logged
+            if (File.Exists(impedFilePath))
+            {
+                File.Delete(impedFilePath);
```

Comment:
why are we deleting previous log if it exists?

Reply:
looks like you're setting the impedFilePath at the start of run time rather than updating it on BICConnect, but you're doing this each connect, so we'd be losing impedance checks from previous connection attemps in the case that there is a reconnect?